source/DamageProfile.h

```diff
@@ -0,0 +1,129 @@
+/* DamageProfile.h
+Copyright (c) 2022 by Amazinite
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+*/
+
+#ifndef DAMAGE_PROFILE_H_
+#define DAMAGE_PROFILE_H_
+
+#include "Point.h"
+#include "Projectile.h"
+
+#include <string>
+
+class Ship;
+class Weapon;
+
+
+// A class that calculates how much damage a ship should take given the ship's
+// attributes and the weapon it was hit by for each damage type.
+class DamageProfile {
+public:
+	class DamageDealt {
```

Comment:
I suppose that would just require not having DamageDealt include the weapon, scale, and isBlast fields (or rather, have DamageDealt include that but extend a class that handles everything else). At that point though if we're using DamageDealt (or some renamed version) for other things, this class might be better placed in Ship instead of here in DamageProfile.

Reply:
Yes, I agree that the class would then be better placed in Ship (or be stand-alone) and should get a more generic name.

Reply:
(I'm not sure how much additional work that is though.)

Reply:
Yup. How does ShipLevels or EnergyLevels sound as a class name? Going from this comment:
https://github.com/endless-sky/endless-sky/blob/3882d31009f52bce40e238db34a4a5970565f796/source/Ship.h#L512-L513

Reply:
ShipLevels or EnergyLevels sounds reasonable as a class name. The class would also describe the damage that a weapon does, the costs of firing a weapon, the costs of using thrusters, the costs of repairs, but those would all be ShipLevel-deltas or EnergyLevel-deltas, so it seems close enough.